lightning/src/ln/channelmanager.rs

```diff
@@ -2059,10 +2128,46 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 		} else if has_err {
 			Err(PaymentSendFailure::AllFailedRetrySafe(results.drain(..).map(|r| r.unwrap_err()).collect()))
 		} else {
-			Ok(())
+			Ok(payment_id)
 		}
 	}
 
+	/// Retries a payment along the given `Route`.
+	///
+	/// Errors returned are a superset of those returned from `send_payment`, so see `send_payment`
+	/// documentation for more details on errors. This method will also error if the retry amount puts
+	/// the payment nontrivially over the payment's total amount, or if the payment for the given
+	/// `payment_id` cannot be found.
```

Comment:
Docs should describe *why* a payment might be missing - timed out or succeeded, presumably is the full list.